jerry-core/ecma/operations/ecma-dataview-object.c

```diff
@@ -314,7 +314,21 @@ ecma_op_dataview_get_set_view_value (ecma_value_t view, /**< the operation's 'vi
     return ecma_get_typedarray_element (swap_block_p, id);
   }
 
-  if (ecma_is_value_number (value_to_set))
+#if ENABLED (JERRY_BUILTIN_BIGINT)
+  bool set_value_is_bigint = ecma_is_value_bigint (value_to_set);
+
+  if (id > ECMA_FLOAT64_ARRAY)
+  {
+    if (!set_value_is_bigint)
```

Comment:
The value should be coerced to a bigint, not just checked. In fact, it seems this whole method needs an update, as regular values should also be coerced to a number, and the coercion should happen much earlier.